packages/typescript-estree/src/parseSettings/createParseSettings.ts

```diff
@@ -127,6 +127,7 @@ export function createParseSettings(
           DEFAULT_TSCONFIG_CACHE_DURATION_SECONDS,
     )),
     tsconfigRootDir,
+    typeAware: !!options.programs || !!options.project,
```

Comment:
I noticed that `hasFullTypeInformation` was checking whether `programs` or `projects` were empty. However, this will always be the case when using the project service. The docs seem to point to using `programs` and `project` (singular) as the way to enable type-aware linting, so I decided to make this explicit here.